api/v1beta1/swift_webhook.go

```diff
@@ -120,7 +125,26 @@ func (r *Swift) ValidateCreate() error {
 func (r *Swift) ValidateUpdate(old runtime.Object) error {
 	swiftlog.Info("validate update", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object update.
+	oldSwift, ok := old.(*Swift)
+	if !ok || oldSwift == nil {
+		return apierrors.NewInternalError(fmt.Errorf("unable to convert existing object"))
+	}
+
+	if *r.Spec.SwiftStorage.Replicas < *oldSwift.Spec.SwiftStorage.Replicas {
+		return apierrors.NewForbidden(
+			schema.GroupResource{
+				Group:    GroupVersion.WithKind("Swift").Group,
+				Resource: GroupVersion.WithKind("Swift").Kind,
+			},
+			r.GetName(),
+			field.Invalid(
+				field.NewPath("spec").Child("swiftStorage").Child("replicas"),
+				*r.Spec.SwiftStorage.Replicas,
+				"SwiftStorage does not support scale-in",
```

Comment:
Minor nit - should this be "scale down" instead of scale-in? To make it more clear that scale up is supported?
Or maybe using the same string as in https://github.com/openstack-k8s-operators/swift-operator/blob/main/controllers/swiftstorage_controller.go#L168?

Reply:
I had an impression scale-up/scale-down increases/decreases resources allocated to pods while scale-out/scale-in increases/decreases number of pods. So I intentionally used scale-in here. though this might not be very common thing.